packages/core/src/modules/credentials/protocol/v2/__tests__/v2-credentials.e2e.test.ts

```diff
@@ -91,7 +100,7 @@ describe('v2 credentials', () => {
     })
 
     testLogger.test('Faber waits for credential proposal from Alice')
-    faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
+    faberCredentialRecord = await waitForCredentialRecordSubject(faberReplay, {
```

Comment:
I've been considering your changes in this test and the test below where 'Faber waits for credential proposal from Alice'.  Here you use **waitForCredentialRecordSubject(faberReplay**.... vs. in the test below where you made the change to create the Promise from **waitForCredentialRecord(faberAgent**.... why the different calls?

Reply:
Okay thank you for clarification! Yes, makes sense.  I'll look through how we query/test for events between the tests so that i can generalize the pattern to other tests.